python/python/lance/dataset.py

```diff
@@ -681,10 +682,21 @@ def version(self) -> int:
     @property
     def latest_version(self) -> int:
         """
-        Returns the lastest version of the dataset
+        Returns the latest version of the dataset.
         """
         return self._ds.latest_version()
 
+    def checkout_version(self, version) -> "LanceDataset":
+        """
+        Load the given version of the dataset.
+
```

Comment:
```suggestion
        Load the given version of the dataset.

        Unlike the :func:`dataset` constructor, this will re-use the
        current cache.
```